core/string.rbs

```diff
@@ -765,7 +765,7 @@ class String
   #
   # Otherwise returns `self.dup`, which is not frozen.
   #
-  def +@: () -> (self | instance)
```

Comment:
`self | instance` type doesn't make much sense.
It will be a `String` instance or it's subclass, so `self` is good here.

Reply:
Did you mean? `instance`